src/server/pachyderm_test.go

```diff
@@ -1113,6 +1113,9 @@ func TestFlushCommit(t *testing.T) {
 		require.NoError(t, err)
 		commitInfos := collectCommitInfos(t, commitIter)
 		require.Equal(t, numStages, len(commitInfos))
+		jobInfos, err := c.FlushJobAll([]*pfs.Commit{client.NewCommit(sourceRepo, commit.ID)}, nil)
+		require.NoError(t, err)
+		require.Equal(t, numStages, len(jobInfos))
```

Comment:
Makes sense. 

I'm not sure what the status of [this issue](https://github.com/pachyderm/pachyderm/issues/2673) is on the invariants branch, but if it is implemented...

What do you think of basing this PR off of the commit invariants branch so that we can add a test that `FlushJob` (and really the underlying `FlushCommit`) returns the expected results when a job fails?

I think if there were a 5 stage pipeline (as in this test) and the 3rd job failed, I'd expect `FlushCommit` to return after that job failed w 2 successful and 3 failed commits. I'd expect `FlushJob` to return (with an error) after that 3rd job failed w 2 successful jobs and 1 failed job. It would be nice to capture these expectations in a test. Especially since the recent customer use cases would use this API in particular to detect failed jobs.